hydrosdk/predictor.py

```diff
@@ -83,31 +83,30 @@ def predict_resp_to_dict_pydtype(response: PredictResponse) -> dict:
 
             dims = [dim.size for dim in tensor_shape.dim]
 
-            output_tensors_dict[key_str] = dims
+            # FIXME value is not extracted from tensor_proto, see  predict_resp_to_dict_nparray
+            output_tensors_dict[key_str] = dims  # FIXME dims is not a value! it is dimenshion shape
         return output_tensors_dict
 
     @staticmethod
     def predict_resp_to_dict_nparray(response: PredictResponse) -> dict:
         output_tensors_dict = {}
-        for key, value in response.outputs.items():
-            key_str = key
-
-            tensor_shape = value.tensor_shape
-
-            dims = [dim.size for dim in tensor_shape.dim]
-
-            dtype = proto2np_dtype(value.dtype)
-            output_tensors_dict[key_str] = np.asarray(dims, dtype=dtype)
+        for tensor_name, tensor_proto in response.outputs.items():
+            array_shape = [dim.size for dim in tensor_proto.tensor_shape.dim]
+            np_dtype = proto2np_dtype(tensor_proto.dtype)
+            value = getattr(tensor_proto, DTYPE_TO_FIELDNAME[tensor_proto.dtype])
+            output_tensors_dict[tensor_name] = np.asarray(value, dtype=np_dtype).reshape(*array_shape if array_shape else (-1,))
 
         return output_tensors_dict
 
     @staticmethod
     def predict_resp_to_df(response: PredictResponse) -> pd.DataFrame:
         df = pd.DataFrame()
-        for key, value in response.outputs.items():
+        for key, tensor_proto in response.outputs.items():
```

Comment:
as you changed previously, let's name key as tensor_name

Reply:
Done